### PR TITLE
[DI] Remove deprecated generating a dumped container without populating the method map

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

See #20113

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 CHANGELOG
 =========
 
+4.0.0
+
+ * [BC BREAK] removed support for generating a dumped `Container` without populating the method map
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md]: BC BREAK prefix not needed

**Comment by nicolas-grekas** [comment]
rebase needed

**Comment by ro0NL** [comment]
Ready!

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.

**Comment by nicolas-grekas** [comment]
Just wondering, should we deprecate the "underscore" method in 3.4?

**Comment by ro0NL** [comment]
it's used a lot still (together with `camelize()` more or less), can have a look soonish.

**Pull request merged**